In [2]:
import pandas as pd
import numpy as np

In [107]:
df = pd.read_csv("data/EMSResponses.csv")
df.head()

incident_ID incident_date  priority            zone    time_phone_pickup  \
0            1      1/1/2012         4  City of Austin  12/31/2011 20:32:54   
1            2      1/1/2012         4  City of Austin  01/01/2012 00:03:46   
2            3      1/1/2012         4  City of Austin  01/01/2012 00:08:43   
3            4      1/1/2012         2  City of Austin  01/01/2012 00:11:47   
4            5      1/1/2012         3  City of Austin  01/01/2012 00:16:14   

     time_unit_arrived time_incident_closed               disposition  \
0  01/01/2012 03:03:07  01/01/2012 03:40:39          ATCEMS Transport   
1  01/01/2012 00:35:56  01/01/2012 01:21:12           Patient Refusal   
2                  NaN  01/01/2012 00:11:29  No Patient / False Alarm   
3  01/01/2012 00:17:22  01/01/2012 01:03:06           Patient Refusal   
4  01/01/2012 00:24:58  01/01/2012 01:09:11          ATCEMS Transport   

   council_district  census_state_code  census_county_code  census_tract  \
0                 9                 48                 453          1100   
1                 5                 48                 453          2428   
2                 9                 48                 453          1100   
3                 3                 48                 453          2314   
4                 2                 48                 453          2430   

   census_block     census_geoID census_block_name  
0          1118  484530011001118        Block 1118  
1          1003  484530024281003        Block 1003  
2          1080  484530011001080        Block 1080  
3          2010  484530023142010        Block 2010  
4          1019  484530024301019        Block 1019

In [108]:
#Remove all of the temporal and non-unique data that we don't care about.
df.drop(["census_state_code", "census_county_code", "census_block_name", "zone"], axis = 1, inplace = True)

In [109]:
df.drop(["incident_date", "time_phone_pickup", "time_unit_arrived","time_incident_closed"], axis = 1, inplace = True)

In [117]:
df["disposition"][df["disposition"] == "<Missing>"] = "Other"

/home/lee/anaconda3/lib/python3.6/site-packages/ipykernel_launcher.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  """Entry point for launching an IPython kernel.


In [118]:
df["census_geoID"] = df["census_geoID"].map(lambda x: int(x/10000))

In [132]:
df.head()
df_EMS = df[df["disposition"] == "ATCEMS Transport"]
df_EMS = df_EMS.groupby("census_geoID").size().reset_index(name = "count")


census_geoID  count
0     48209010901      6
1     48453000101   2761
2     48453000102    480
3     48453000203   2298
4     48453000204   2595
5     48453000205   1474
6     48453000206   1590
7     48453000302   2171
8     48453000304   1110
9     48453000305   1041
10    48453000306   2211
11    48453000307   1321
12    48453000401   2368
13    48453000402   2365
14    48453000500    563
15    48453000601   2429
16    48453000603   1414
17    48453000604   1529
18    48453000700   3537
19    48453000801   1239
20    48453000802   2982
21    48453000803   1729
22    48453000804   2386
23    48453000901   1156
24    48453000902   6507
25    48453001000   2860
26    48453001100  34465
27    48453001200   1663
28    48453001303   2273
29    48453001304   1255
..            ...    ...
185   48453002422   2397
186   48453002423   1135
187   48453002424   2421
188   48453002425    470
189   48453002426   1252
190   48453002427   3183
191   48453002428   1024
192   48453002429    518
193   48453002430    843
194   48453002431   4708
195   48453002432    129
196   48453002433   1076
197   48453002434     20
198   48453002435    876
199   48453002500   2852
200   48453980000   1808
201   48491020311    985
202   48491020316      7
203   48491020318    673
204   48491020403    791
205   48491020404    737
206   48491020405   1594
207   48491020406   2402
208   48491020408   1234
209   48491020409    295
210   48491020410   1314
211   48491020411     89
212   48491020508    757
213   48491020509    265
214   48491020510    239

[215 rows x 2 columns]

In [180]:
df_population = pd.read_csv("data/PopulationData.csv")
df_population = df_population[["GEO.id2", "HC01_VC03"]]
df_population = df_population.iloc[1:].astype(int)

In [181]:
df_EMS.merge(df_population, left_on = "census_geoID", right_on = "GEO.id2", how = 'inner')

census_geoID  count      GEO.id2  HC01_VC03
0     48453000101   2761  48453000101       4061
1     48453000102    480  48453000102       2556
2     48453000203   2298  48453000203       1460
3     48453000204   2595  48453000204       3092
4     48453000205   1474  48453000205       4154
5     48453000206   1590  48453000206       3251
6     48453000302   2171  48453000302       5111
7     48453000304   1110  48453000304       3359
8     48453000305   1041  48453000305       3272
9     48453000306   2211  48453000306       3603
10    48453000307   1321  48453000307       2009
11    48453000401   2368  48453000401       3515
12    48453000402   2365  48453000402       3271
13    48453000500    563  48453000500       4894
14    48453000601   2429  48453000601       9792
15    48453000603   1414  48453000603       7672
16    48453000604   1529  48453000604       6839
17    48453000700   3537  48453000700       1076
18    48453000801   1239  48453000801       1638
19    48453000802   2982  48453000802       3494
20    48453000803   1729  48453000803       2333
21    48453000804   2386  48453000804       2300
22    48453000901   1156  48453000901       2165
23    48453000902   6507  48453000902       5031
24    48453001000   2860  48453001000       3687
25    48453001100  34465  48453001100       5331
26    48453001200   1663  48453001200       4907
27    48453001303   2273  48453001303       3289
28    48453001304   1255  48453001304       3769
29    48453001305   3635  48453001305       5769
..            ...    ...          ...        ...
170   48453002316   2686  48453002316       5405
171   48453002317   1663  48453002317       5150
172   48453002318   3093  48453002318      10290
173   48453002319   2471  48453002319       2264
174   48453002402   2895  48453002402       7329
175   48453002403   1965  48453002403       2575
176   48453002407   1356  48453002407       5896
177   48453002409    987  48453002409       3529
178   48453002410   1240  48453002410       4132
179   48453002411   2897  48453002411       6858
180   48453002412   1469  48453002412       6082
181   48453002413   1999  48453002413       4953
182   48453002419   1497  48453002419       4327
183   48453002421   3294  48453002421       9346
184   48453002422   2397  48453002422       5511
185   48453002423   1135  48453002423       5690
186   48453002424   2421  48453002424       3871
187   48453002425    470  48453002425       3536
188   48453002426   1252  48453002426       9702
189   48453002427   3183  48453002427       8261
190   48453002428   1024  48453002428       6618
191   48453002429    518  48453002429        762
192   48453002430    843  48453002430       3482
193   48453002431   4708  48453002431       8167
194   48453002432    129  48453002432       2087
195   48453002433   1076  48453002433       6793
196   48453002434     20  48453002434       1764
197   48453002435    876  48453002435       8019
198   48453002500   2852  48453002500       5630
199   48453980000   1808  48453980000          0

[200 rows x 4 columns]

In [120]:
df.to_csv("ReducedEMSResponse.csv")

In [75]:
df2 = df.groupby("disposition").size().reset_index(name="total_incidents")
df2["average_per_month"] = (df2["total_incidents"] / 58).round()
df2["average_per_year"] = df2["average_per_month"] * 12
df2

disposition  total_incidents  average_per_month  \
0          ATCEMS Transport           381823             6583.0   
1                 Cancelled            43441              749.0   
2         Deceased on Scene             4814               83.0   
3  No Patient / False Alarm            58220             1004.0   
4                     Other             1312               23.0   
5           Patient Refusal            95381             1644.0   
6                  Referred            35332              609.0   
7                      Test               29                0.0   

   average_per_year  
0           78996.0  
1            8988.0  
2             996.0  
3           12048.0  
4             276.0  
5           19728.0  
6            7308.0  
7               0.0

Between 01/01 2012 and 10/31/2017, ATCEMS transport was dispatched to 381823 incidents

# Prevalence Data

In [139]:
df3 = pd.read_csv("data/AdultsArthritis.csv")
df3 = df3[["Location", "Indicator Value"]]
df3=df3.rename(columns = {'Indicator Value':'Arthritis'})

In [140]:
files = ["data/AdultsCancer.csv", "data/AdultsDiabetes.csv", "data/AdultsHighCholesterol.csv", "data/AdultsObese.csv",\
         "data/HighBloodPressure.csv", "data/AdultsRoutineCheckup.csv", "data/AdultsWithoutHealthInsurance.csv",\
         "data/ElderlyBelowPovertyLevel.csv", "data/ElderlyLivingAlone.csv",\
         "data/ElderlyRecievedRecommendedPreventationServicesFemales.csv","data/ElderlyRecievedRecommendedPreventativeServicesMales.csv",\
         "data/HouseholdsWithoutVehicle.csv"]

labels = ["Cancer", "Diabetes", "Cholesterol", "Obese", "HBP", "Checkup", "NoInsurance", "Poverty", "Alone",\
          "FemaleServices", "MaleServices", "NoVehicles"]

df2 = pd.read_csv("data/AdultsCancer.csv")
df2 = df2[["Location", "Indicator Value"]]
df2=df2.rename(columns = {'Indicator Value':'Cancer'})
df3 = df3.merge(df2, on = "Location")

df2 = pd.read_csv("data/AdultsDiabetes.csv")
df2 = df2[["Location", "Indicator Value"]]
df2=df2.rename(columns = {'Indicator Value':'Diabetes'})
df3 = df3.merge(df2, on = "Location")

df2 = pd.read_csv("data/AdultsHighCholesterol.csv")
df2 = df2[["Location", "Indicator Value"]]
df2=df2.rename(columns = {'Indicator Value':'Cholesterol'})
df3 = df3.merge(df2, on = "Location")

df2 = pd.read_csv("data/AdultsObese.csv")
df2 = df2[["Location", "Indicator Value"]]
df2=df2.rename(columns = {'Indicator Value':'Obese'})
df3 = df3.merge(df2, on = "Location")

df2 = pd.read_csv("data/HighBloodPressure.csv")
df2 = df2[["Location", "Indicator Value"]]
df2=df2.rename(columns = {'Indicator Value':'HBP'})
df3 = df3.merge(df2, on = "Location")

df2 = pd.read_csv("data/AdultsRoutineCheckup.csv")
df2 = df2[["Location", "Indicator Value"]]
df2=df2.rename(columns = {'Indicator Value':'Routine'})
df3 = df3.merge(df2, on = "Location")

df2 = pd.read_csv("data/AdultsWithoutHealthInsurance.csv")
df2 = df2[["Location", "Indicator Value"]]
df2=df2.rename(columns = {'Indicator Value':'NoInsurance'})
df3 = df3.merge(df2, on = "Location")

df2 = pd.read_csv("data/ElderlyBelowPovertyLevel.csv")
df2 = df2[df2["Period of Measure"] == "2011-2015"]
df2 = df2[["Location", "Indicator Value"]]
df2=df2.rename(columns = {'Indicator Value':'ElderlyPoverty'})
df3 = df3.merge(df2, on = "Location")

df2 = pd.read_csv("data/ElderlyLivingAlone.csv")
df2 = df2[df2["Period of Measure"] == "2011-2015"]
df2 = df2[["Location", "Indicator Value"]]
df2=df2.rename(columns = {'Indicator Value':'ElderlyAlone'})
df3 = df3.merge(df2, on = "Location")

df2 = pd.read_csv("data/ElderlyRecievedRecommendedPreventationServicesFemales.csv")
df2 = df2[["Location", "Indicator Value"]]
df2=df2.rename(columns = {'Indicator Value':'FemaleServices'})
df3 = df3.merge(df2, on = "Location")

df2 = pd.read_csv("data/ElderlyRecievedRecommendedPreventativeServicesMales.csv")
df2 = df2[["Location", "Indicator Value"]]
df2=df2.rename(columns = {'Indicator Value':'MaleServices'})
df3 = df3.merge(df2, on = "Location")

df2 = pd.read_csv("data/HouseholdsWithoutVehicle.csv")
df2 = df2[df2["Period of Measure"] == "2011-2015"]
df2 = df2[["Location", "Indicator Value"]]
df2=df2.rename(columns = {'Indicator Value':'NoVehicle'})
df3 = df3.merge(df2, on = "Location")

In [141]:
df3

Location  Arthritis  Cancer  Diabetes  Cholesterol  Obese   HBP  \
0    48453000101       17.4     7.3       6.5         35.8   19.3  26.9   
1    48453000102       20.5     8.8       7.8         41.9   19.8  32.0   
2    48453000203        6.9     2.3       3.2         22.6   17.8  14.3   
3    48453000204       12.0     4.1       5.1         30.9   20.9  20.2   
4    48453000205       11.1     3.8       5.1         29.4   21.6  19.3   
5    48453000206       16.8     6.3       7.1         36.2   21.6  26.7   
6    48453000302       10.0     3.4       4.5         28.2   20.0  18.0   
7    48453000304        9.1     2.9       4.5         27.4   21.3  18.2   
8    48453000305        9.4     3.2       4.2         26.8   20.7  17.0   
9    48453000306       10.3     3.3       5.3         28.1   22.6  18.7   
10   48453000307       15.0     4.9       7.2         33.9   24.2  25.0   
11   48453000401        8.3     2.8       3.9         25.4   18.5  16.1   
12   48453000402       14.7     4.6       8.3         31.6   24.0  26.1   
13   48453000500        8.1     2.6       3.7         24.5   19.6  15.2   
14   48453000603        4.3     1.0       2.2         14.9   20.0  11.1   
15   48453000700       12.2     4.5       5.8         33.2   19.4  23.4   
16   48453000801       22.7     5.4      17.5         41.5   32.3  34.8   
17   48453000802       19.3     4.0      14.1         36.1   35.4  33.8   
18   48453000803       13.2     3.6       8.2         30.2   26.5  24.3   
19   48453000804       17.4     4.0      12.2         35.2   32.5  31.1   
20   48453000901       11.5     3.1       7.7         29.8   25.7  22.7   
21   48453000902       16.3     3.7      12.6         37.6   31.2  28.9   
22   48453001000       18.0     4.8      12.7         38.3   28.2  29.1   
23   48453001100       11.4     4.1       5.7         31.3   21.8  21.9   
24   48453001200       10.1     4.1       4.1         29.1   18.4  18.7   
25   48453001303       14.0     5.2       5.8         34.2   21.5  23.5   
26   48453001304       12.6     4.6       5.2         31.3   21.5  21.0   
27   48453001305       12.5     3.9       6.5         31.8   23.9  22.0   
28   48453001307       13.7     4.0       7.8         33.2   25.9  23.5   
29   48453001308       13.9     4.0       8.2         32.4   25.5  22.4   
..           ...        ...     ...       ...          ...    ...   ...   
155  48453002313       10.8     2.6       6.8         28.0   27.6  19.1   
156  48453002314       10.1     2.1       7.0         27.7   29.9  20.3   
157  48453002315       11.0     2.3       8.2         29.0   31.2  20.2   
158  48453002316       11.5     2.1       9.3         30.1   33.6  21.9   
159  48453002317        5.8     1.2       4.3         20.0   24.3  14.6   
160  48453002318       11.2     2.2       8.4         29.0   28.9  20.3   
161  48453002402       12.6     3.4       7.7         31.6   25.6  21.5   
162  48453002403       15.5     4.4       9.2         34.7   26.6  24.0   
163  48453002407       13.4     4.3       6.9         31.7   23.4  20.8   
164  48453002409       17.0     4.4      10.7         36.9   29.3  27.5   
165  48453002410       15.4     3.8       9.8         34.4   29.5  24.7   
166  48453002411       15.8     2.8      13.1         35.1   37.1  26.2   
167  48453002412       15.7     3.1      12.8         34.9   34.2  24.8   
168  48453002413       14.7     2.5      12.2         33.4   37.0  25.5   
169  48453002419       10.1     2.2       7.0         27.7   29.2  19.3   
170  48453002421       10.7     3.3       5.7         28.5   23.3  18.5   
171  48453002422       15.4     4.1       9.4         34.8   28.2  24.7   
172  48453002423       15.3     4.2       8.8         35.1   27.4  25.2   
173  48453002424       17.3     4.7      10.2         36.6   28.5  26.9   
174  48453002425       10.5     3.2       6.0         28.7   23.9  18.7   
175  48453002426       11.7     2.8       8.2         30.0   29.2  20.6   
176  48453002427       15.6     3

In [105]:
df3.to_csv("IndicatorsData.csv")

In [154]:
df_Final = df3.merge(df_EMS[["census_geoID", "count"]], left_on = "Location", right_on = "census_geoID", how = "inner")

In [153]:
df_Final.to_csv("CleanedDataset")

array([  4.84530001e+10,   4.84530001e+10,   4.84530002e+10,
         4.84530002e+10,   4.84530002e+10,   4.84530002e+10,
         4.84530003e+10,   4.84530003e+10,   4.84530003e+10,
         4.84530003e+10,   4.84530003e+10,   4.84530004e+10,
         4.84530004e+10,   4.84530005e+10,   4.84530006e+10,
         4.84530007e+10,   4.84530008e+10,   4.84530008e+10,
         4.84530008e+10,   4.84530008e+10,   4.84530009e+10,
         4.84530009e+10,   4.84530010e+10,   4.84530011e+10,
         4.84530012e+10,   4.84530013e+10,   4.84530013e+10,
         4.84530013e+10,   4.84530013e+10,   4.84530013e+10,
         4.84530014e+10,   4.84530014e+10,   4.84530014e+10,
         4.84530015e+10,   4.84530015e+10,   4.84530015e+10,
         4.84530015e+10,   4.84530016e+10,   4.84530016e+10,
         4.84530016e+10,   4.84530016e+10,   4.84530017e+10,
         4.84530017e+10,   4.84530017e+10,   4.84530017e+10,
         4.84530017e+10,   4.84530017e+10,   4.84530017e+10,
         4.84530017e+10,

In [146]:
df_Final.drop("census_geoID", axis = 1, inplace = True)

In [155]:
#df_Final is something that we can actually use for a clustering analysis.